search/includes/classes/class-health.php

```diff
@@ -176,44 +193,42 @@ public static function validate_index_posts_content( $start_post_id = 1, $last_p
 		}
 
 		do {
-			$next_batch_post_id = $start_post_id + self::CONTENT_VALIDATION_BATCH_SIZE;
+			$next_batch_post_id = $start_post_id + $batch_size;
 
 			if ( $last_post_id < $next_batch_post_id ) {
 				$next_batch_post_id = $last_post_id + 1;
 			}
 
-			if ( $is_cli ) {
-				\WP_CLI::line( sprintf( 'Validating posts %d - %d', $start_post_id, $next_batch_post_id - 1 ) );
+			if ( $is_cli && ! $silent ) {
+				echo sprintf( 'Validating posts %d - %d', $start_post_id, $next_batch_post_id - 1 ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 			}
 			
 			$result = self::validate_index_posts_content_batch( $indexable, $start_post_id, $next_batch_post_id );
 
 			if ( is_wp_error( $result ) ) {
-				$result = [
-					'entity' => $indexable->slug,
-					'start_post_id' => $start_post_id,
-					'error' => $result->get_error_message(),
-				];
+				$result['errors'] = array( sprintf( 'batch %d - %d (entitiy: %s) error: %s', $start_post_id, $next_batch_post_id - 1, $indexable->slug, $result->get_error_message() ) );
 			}
 
 			$results = array_merge( $results, $result );
 
 			// Limit $results size
-			if ( strlen( serialize( $results ) ) > self::CONTENT_VALIDATION_MAX_DIFF_SIZE ) {
-				$error = new WP_Error( 'diff-size-limit-reached', sprintf( 'Reached diff size limit of %d bytes, aborting', self::CONTENT_VALIDATION_MAX_DIFF_SIZE ) );
+			if ( strlen( serialize( $results ) ) > $max_diff_size ) {
```

Comment:
The serialize is probably pretty slow - we can switch this out to not limit on byte size, but instead on count of inconsistencies, which would be a super fast check on `count( $results )`.